docs/docs/documentation/components/proxies.md

```diff
@@ -0,0 +1,473 @@
+| **Task Initiation** | Can only receive tasks from mesh agents | Can initiate tasks to any agent |
+| **Use Case** | External/third-party agents | Agents you control |
+
+## Architecture Overview
+
+The proxy sits between the Solace event mesh and external A2A agents, performing bidirectional protocol translation:
+
+```mermaid
+graph LR
+    A[Agent Mesh<br/>A2A-over-Solace] <-->|Solace Topics| B[Proxy Component]
+    B <-->|HTTPS| C[External Agent 1<br/>A2A-over-HTTPS]
+    B <-->|HTTPS| D[External Agent 2<br/>A2A-over-HTTPS]
+    B <-->|HTTPS| E[External Agent N<br/>A2A-over-HTTPS]
+    
+    style B fill:none,stroke:#00C895,stroke-width:2px
+    style A fill:none,stroke:#333,stroke-width:2px
+    style C fill:none,stroke:#333,stroke-width:2px
+    style D fill:none,stroke:#333,stroke-width:2px
+    style E fill:none,stroke:#333,stroke-width:2px
+```
+
+The proxy performs these operations:
+
+1. **Request Flow**: Receives A2A requests from the mesh, resolves artifact URIs to byte content, forwards HTTPS requests to external agents, and streams responses back to the mesh.
+
+2. **Response Flow**: Receives responses from external agents, saves artifacts to the mesh's artifact service, replaces byte content with artifact URIs, and publishes responses to mesh topics.
+
+3. **Discovery Flow**: Periodically fetches agent cards from external agents, updates the local registry, and publishes cards to the mesh discovery topic.
+
+## Configuration
+
+Proxies are configured through YAML files that specify the namespace, downstream agents, authentication, and service settings.
+
+### Basic Configuration
+
+A single proxy can manage multiple external agents. Each agent in the `proxied_agents` list can have its own URL, authentication, and timeout configuration:
+
+```yaml
+app:
+  class_name: solace_agent_mesh.agent.proxies.a2a.app.A2AProxyApp
+  name: my-a2a-proxy
+  app_config:
+    namespace: "myorg/production"
+    proxied_agents:
+      - name: "external-data-agent"
+        url: "https://api.example.com/agent"
+        request_timeout_seconds: 120
+      - name: "external-analytics-agent"
+        url: "https://analytics.example.com/agent"
+        request_timeout_seconds: 180
+      - name: "external-reporting-agent"
+        url: "https://reports.example.com/agent"
+    artifact_service:
+      type: "filesystem"
+      base_path: "/tmp/proxy-artifacts"
+    discovery_interval_seconds: 60
+    default_request_timeout_seconds: 300
+
+broker:
+  # Broker configuration inherited from environment or specified here
+```
+
+### Configuration Parameters
+
+- `namespace`: The topic prefix for A2A communication (for example, "myorg/production").
+- `proxied_agents`: A list of external agents to proxy. Each agent can have its own URL, authentication, and timeout settings (see Authentication Types below).
+- `artifact_service`: Configuration for storing artifacts (memory, filesystem, or GCS). This is shared across all proxied agents.
```

Comment:
Your example shows `base_path` for the filesystem option. Are there any additional parameters for the other options?